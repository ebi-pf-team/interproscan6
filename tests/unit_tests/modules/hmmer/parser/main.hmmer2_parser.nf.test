nextflow_process {

    name "Test Process HMMER2_PARSER"
    script "interproscan/modules/hmmer/parser/main.nf"
    process "HMMER2_PARSER"

    stage {
        symlink "tests/unit_tests/test_inputs/hmmer2_parser/fasta.faa"
        symlink "tests/unit_tests/test_inputs/hmmer2_parser/hmmpfam.out"
        symlink "interproscan/scripts/hmmer/parse_hmmpfam_out.py"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer2_parser/hmmpfam.out")
                input[1] = file("meta/tests/unit_tests/test_inputs/hmmer2_parser/fasta.faa")
                input[2] = "smart"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
