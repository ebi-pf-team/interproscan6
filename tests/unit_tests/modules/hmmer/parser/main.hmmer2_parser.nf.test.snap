{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "hmmer_parsed_hmmpfam.out.json:md5,f5429c76916bd6ad4f6f5ca5fc97fd62"
                ],
                "1": [
                    "fasta.faa:md5,562770e49f4cc1ce7db3e7ce48d8d23d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T08:54:18.982419"
    }
}